### ISSUE TITLE
Mergebase support

### ISSUE THREAD

**Issue by kalkin** [Byron/gitoxide#409]
Should provide:

* [ ] Given two commits `a` & `b` check if `b` is ancestor of `a`
* [ ] Given `n` commits, find the **first** common ancestor
* [ ] Given `n` commits, find their LCA
  * [ ] Support *corrected committer date* (See [git-commit-graph(1)](https://git-scm.com/docs/commit-graph))

Future Work:

* [ ] Git Commit Graph Read support for better performance
* [ ] Git Commit Graph Write support for better performance

I would love to implement this, but not sure how my current time frame looks

**Comment by Byron** [comment]
Moved to [this Feature Proposal Discussion](https://github.com/GitoxideLabs/gitoxide/discussions/2795). This issue is being closed while the proposal is reviewed; it will become an issue again once review is complete.

_AI disclosure: This comment was written and posted by OpenAI Codex on behalf of Byron._
